jung-visualization/src/test/java/edu/uci/ics/jung/visualization/sublayout/GraphCollapserTest.java

```diff
@@ -0,0 +1,134 @@
+package edu.uci.ics.jung.visualization.sublayout;
+
+import com.google.common.collect.Sets;
+import com.google.common.graph.EndpointPair;
+import com.google.common.graph.MutableNetwork;
+import com.google.common.graph.Network;
+import com.google.common.graph.NetworkBuilder;
+import edu.uci.ics.jung.visualization.picking.MultiPickedState;
+import edu.uci.ics.jung.visualization.subLayout.GraphCollapser;
+import java.util.Collection;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Created by tanelso on 9/26/17. */
+public class GraphCollapserTest {
+
+  Logger log = LoggerFactory.getLogger(GraphCollapserTest.class);
+
+  @Test
+  public void testCollapser() {
+    Network network = getDemoGraph();
+
+    Assert.assertEquals(network.nodes(), Sets.newHashSet("A", "B", "C"));
+    Assert.assertEquals(network.incidentNodes(0), EndpointPair.unordered("B", "A"));
+    Assert.assertEquals(network.incidentNodes(1), EndpointPair.unordered("C", "A"));
+    Assert.assertEquals(network.incidentNodes(2), EndpointPair.unordered("B", "C"));
+
+    GraphCollapser collapser = new GraphCollapser(network);
+    MultiPickedState picker = new MultiPickedState();
+    picker.pick("B", true);
+    picker.pick("C", true);
+
+    Network clusterGraph = collapser.getClusterGraph(network, picker.getPicked());
+    Network collapsed = collapser.collapse(network, clusterGraph);
+    for (Object node : collapsed.nodes()) {
+      if (node instanceof Network) {
+        Assert.assertEquals(((Network) node).edges(), Sets.newHashSet(2));
+      } else {
+        Assert.assertEquals(node, "A");
+      }
+    }
+
+    Assert.assertEquals(collapsed.edges(), Sets.newHashSet(0, 1));
+    for (Object edge : collapsed.edges()) {
+      Assert.assertEquals(collapsed.incidentNodes(edge).nodeU(), "A");
+      Assert.assertTrue(collapsed.incidentNodes(edge).nodeV() instanceof Network);
+    }
+
+    Collection nodes = collapsed.nodes();
+    picker.clear();
+    for (Object node : collapsed.nodes()) {
+      if (node instanceof Network) {
+        picker.pick(node, true);
+      }
+    }
+    Network expanded = collapser.expand(network, collapsed, clusterGraph);
+    Assert.assertEquals(network.nodes(), Sets.newHashSet("A", "B", "C"));
+    Assert.assertEquals(expanded.incidentNodes(0), EndpointPair.unordered("B", "A"));
+    Assert.assertEquals(expanded.incidentNodes(1), EndpointPair.unordered("C", "A"));
+    Assert.assertEquals(expanded.incidentNodes(2), EndpointPair.unordered("B", "C"));
+  }
+
+  @Test
+  public void testTwoConnectedClustersExpandOneThenTheOther() {
+    Network originalNetwork = getDemoGraph2();
+    GraphCollapser collapser = new GraphCollapser(originalNetwork);
+    MultiPickedState picker = new MultiPickedState();
+    picker.pick("A", true);
+    picker.pick("B", true);
+    picker.pick("C", true);
+
+    log.debug("originalNetwork:" + originalNetwork);
```

Comment:
Again, I'm skeptical of the value of logging like this, when IMO stdout would suffice (as well as asserting on what sort of contents and shape you expect `originalNetwork` to have at this stage).

Reply:
Sometimes I like to see which thread and what time something happened, and what class spewed the message. I confess that I have written System.err.println many times in my past though.
You won't see these messages when you build unless you turn on logging. I'd rather have a real unit test that can be quiet than always having console output from who-knows-which test.

Did you see the assertions later in the test?